src/debug/ee/controller.cpp

```diff
@@ -4977,7 +4979,16 @@ DebuggerBreakpoint::DebuggerBreakpoint(Module *module,
             pGenericInstanceFilter = nativeMethodDesc;
         }
     }
-#endif
+#endif // DEBUG
+
+    // This is a special mode that sets a breakpoint at native offset 0 on all current and future native code bodies. 
+    // It's indicated by setting nativeCodeBindAllVersions and native=FALSE (i.e. an IL breakpoint).
+    if (nativeCodeBindAllVersions && !native && offset == 0)
```

Comment:
This should be removed, an IL breakpoint at offset 0 will meet all these conditions and be incorrectly transformed into a native breakpoint at offset 0. Instead if you make the adjustments on the DBI side then this won't be necessary.